controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -875,6 +973,14 @@ bool tlvPrefilledData::add_tlvs_from_list(std::list<wfa_ca::tlv_hex_t> &tlv_hex_
 
         auto values = string_utils::str_split(*tlv.value, ' ');
         for (auto value : values) {
+            //Do conversion if needed
+            if (net::network_utils::is_valid_mac(value)) {
+                mac_to_hex_notation(value);
+            } else if (validate_binary_format(value)) {
+                binary_to_hex(value);
+            } else if (validate_decimal_format(value)) {
+                decimal_to_hex(value);
+            }
```

Comment:
I think it's pretty pointless to convert the string to another string and then convert it to uint8_t, while it is easier to directly convert it to uint8_t... So instead of mac_to_hex_notation, I think it's better to add a member function `tlvPrefilledData::add_tlv_value_mac(value)` that directly writes to `m_buff_ptr__`. The update of `length` can be done either by passing it as in-out, or by returning the size (i.e. calling it as `length -= add_tlv_value_mac(value)`).

Reply:
I may not have been sufficiently clear in my earlier remark... I only mentioned `add_tlv_value_mac`, but I meant the same thing for the decimal and binary cases.

Reply:
I thought so but I wasn't sure, I'll add that.